src/background.js

```diff
@@ -251,28 +284,14 @@ const rollout = {
     // If the heuristics say to enable DoH, determine if the doorhanger
     // should be shown
     } else if (shouldShowDoorhanger) {
-      browser.experiments.doorhanger.onDoorhangerAccept.addListener(
-        rollout.doorhangerAcceptListener
-      );
-      browser.experiments.doorhanger.onDoorhangerDecline.addListener(
-        rollout.doorhangerDeclineListener
-      );
-      await browser.experiments.doorhanger.show({
-        name: browser.i18n.getMessage("doorhangerName"),
-        text: "<> " + browser.i18n.getMessage("doorhangerBody"),
-        okLabel: browser.i18n.getMessage("doorhangerButtonOk"),
-        okAccessKey: browser.i18n.getMessage("doorhangerButtonOkAccessKey"),
-        cancelLabel: browser.i18n.getMessage("doorhangerButtonCancel"),
-        cancelAccessKey: browser.i18n.getMessage("doorhangerButtonCancelAccessKey"),
-      });
-
-      await stateManager.rememberDoorhangerShown();
-
-    // If the doorhanger doesn't need to be shown and the heuristics
-    // say to enable DoH, enable it
+      await stateManager.showDoorHangerAndEnableDoH();
```

Comment:
Rolled this function (as it is now used twice) into its own function. Robust function names!